### PR TITLE
Expand houses offered for sale

### PR DESCRIPTION
**Pull request by ajrb**
Doesn't match classic exactly.
Fixed house ownership to work with new discovery system, at present if players buy a house it's not shown on the map.

### CODE REVIEW

File: Assets/Scripts/Game/BuildingDirectory.cs [hunk 1]
```diff
@@ -157,11 +155,31 @@ public List<BuildingSummary> GetBuildingsOfFaction(int factionId)
...
+            int maxForSale = Mathf.Min(buildingDict.Count / 10, 20);
             List<BuildingSummary> forSale = new List<BuildingSummary>();
+            List<BuildingSummary> candidates = new List<BuildingSummary>();
             foreach (BuildingSummary building in buildingDict.Values)
+            {
                 if (building.BuildingType == DFLocation.BuildingTypes.HouseForSale)
+                {
                     forSale.Add(building);
+                }
+                else if (building.BuildingType > DFLocation.BuildingTypes.House1 &&
```
Comment (df-ferital) [inline Assets/Scripts/Game/BuildingDirectory.cs]: Should be >= to include House1 type

File: Assets/Scripts/Game/BuildingDirectory.cs [hunk 2]
```diff
@@ -157,11 +155,31 @@ public List<BuildingSummary> GetBuildingsOfFaction(int factionId)
...
+                         building.BuildingType < DFLocation.BuildingTypes.House4 &&
```
Comment (df-ferital) [inline Assets/Scripts/Game/BuildingDirectory.cs]: Should be <= to include House4 type

Comment (ajrb) [reply Assets/Scripts/Game/BuildingDirectory.cs]: Good catch. Copied from an exclusion conditional and brain farted that switch around.

File: Assets/Scripts/Game/BuildingDirectory.cs (hunk 1 above)
Comment (ajrb) [reply Assets/Scripts/Game/BuildingDirectory.cs]: Good catch. Copied from an exclusion conditional and brain farted that switch around.

**Pull request merged**